resolwe/flow/views/annotations.py

```diff
@@ -75,25 +83,33 @@ class AnnotationValueViewSet(
     queryset = AnnotationValue.objects.all()
     permission_classes = (get_permissions_class(),)
 
-    def _get_entity(self, request: request.Request) -> AnnotationValue:
-        """Get annotation value from request.
-
-        :raises ValidationError: if the annotation value is not valid.
-        :raises NotFound: if the user is not authenticated.
+    def get_serializer(self, *args, **kwargs):
         """
-        serializer = self.get_serializer(data=request.data, partial=True)
-        serializer.is_valid(raise_exception=True)
-        return serializer.validated_data["entity"]
+        Return the serializer instance that should be used for validating and
+        deserializing input, and for serializing output.
+        """
+        kwargs["many"] = isinstance(self.request.data, list)
+        return super().get_serializer(*args, **kwargs)
 
     def create(self, request, *args, **kwargs):
-        """Create annotation value.
+        """Create annotation value(s).
 
-        Authenticated users with edit permissions on the entity can create annotations.
+        Create one or more annotation values. The request data can be either dict or
+        list (in case of bulk create).
         """
-        entity = self._get_entity(request)
-        if entity.has_permission(Permission.EDIT, request.user):
-            return super().create(request, *args, **kwargs)
-        elif entity.has_permission(Permission.VIEW, request.user):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        raw_data = serializer.validated_data
+        validated_data = raw_data if isinstance(raw_data, list) else [raw_data]
+        entities = {value["entity"] for value in validated_data}
+        # Check permissions on entities.
+        if not all(
+            entity.has_permission(Permission.EDIT, request.user) for entity in entities
+        ):
             raise exceptions.PermissionDenied()
-        else:
-            raise exceptions.NotFound()
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return response.Response(
+            serializer.data, status=status.HTTP_201_CREATED, headers=headers
```

Comment:
What will be the response for the singe object? The object itself or a list with one element?

Reply:
Old.